### PR TITLE
Dev 2125 feat/add password regex option

### PR DESCRIPTION
**Pull request by Achalhii**
In this feat, we add a new password regexp option in the auth service.

The routes concerned are : 
- signup
- reset-password
- updatePassword

btw, a verification of the email field has been added

Tested locally with caas-api & caas-styleguide

Doc updated

### CODE REVIEW

**Review by jlotery — CHANGES_REQUESTED** [review]
Can you please create some tests in the tests/auth folder that prove that the regex test works i.e. create a user that that has a password that you know will fail and one that you know will work and test the return reuslt

File: services/auth/lib/local.js [hunk 1]
```diff
@@ -111,6 +111,13 @@ module.exports.createRandomToken = function (username, salt) {
 
 module.exports.signup = function (req, res) {
   const salt = req.authProvider.options.salt;
+  const passwordRegex = new RegExp(req.authProvider.options.passwordRegex);
```
Comment (gabila-r) [inline services/auth/lib/local.js:114]: What happens if `passwordRegex` is not in the config ?

**Review by jlotery — CHANGES_REQUESTED** [review]
In the tests can you also add a test for an invalid email address with as many edge cases as possible. Thanks :)

File: services/auth/lib/local.js (hunk 1 above)
Comment (Achalhii) [reply services/auth/lib/local.js:114]: if passwordRegex is undefined, automatically take '' as pattern. You can test it, however, I can't find any doc that explicitly states it. 
Then, I will therefore add security in my code, you never know!

**Review by jlotery — APPROVED** [review]
LGTM
